src/basic/handlers/basic.ts

```diff
@@ -229,6 +237,48 @@ class BasicStepHandler implements FRUIStepHandler {
       return x.getOptions && x.getOptions().length === 1;
     });
 
+    /**
+     * A SelectIdPCallback with local authentication form is a special combo
+     */
+    const socialLoginWithLocal = (): boolean => {
+      let hasSelectIdP = false;
+      let hasLocalForm = false;
+
+      callbacks.forEach((x: FRCallback) => {
+        switch (x.getType()) {
+          case CallbackType.SelectIdPCallback:
+            hasSelectIdP = true;
+            break;
+          case CallbackType.NameCallback:
+            hasLocalForm = true;
+            break;
+          default:
+          // Intentionally left empty
+        }
+      });
+
+      return hasSelectIdP && hasLocalForm;
+    };
+
+    if (socialLoginWithLocal() && rendererClassname !== 'SelectIdPCallbackRenderer') {
+      /**
+       * If Local Authentication form is combined with Social Login Providers, and
+       * local inputs are used, this signals the use of the local auth, so set
+       * localAuthentication as provider on the SelectIdPCallback before resolving
+       */
+      const selectIdPRenderer = this.renderers.find((x) => {
+        return x.constructor.name === 'SelectIdPCallbackRenderer';
+      });
+      // eslint-disable-next-line
+      // @ts-ignore
+      selectIdPRenderer.onOtherInput('localAuthentication');
+    }
```

Comment:
This is all the logic required to appropriate detect whether the step is just a list of provider buttons or a mix of buttons and a local authentication form. The latter requires special handling for form validation and submission.